novawallet/Modules/PolkadotVault/WalletScanner/PVScanInterator.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+import Operation_iOS
+
+enum PVScanInteractorError: Error {
+    case invalidAddress
+    case invalidChain
+}
+
+final class PVScanInteractor {
+    weak var presenter: PVScanInteractorOutputProtocol?
+
+    let chainRegistry: ChainRegistryProtocol
+
+    init(chainRegistry: ChainRegistryProtocol) {
+        self.chainRegistry = chainRegistry
+    }
+}
+
+extension PVScanInteractor: PVScanInteractorInputProtocol {
+    func process(accountScan: PolkadotVaultAccount) {
+        do {
+            let chainId = accountScan.genesisHash.toHex()
+
+            // make sure that genesis hash is from valid chain
+
+            guard let chain = chainRegistry.getChain(for: chainId), !chain.isEthereumBased else {
```

Comment:
I think we shouldn't care from which chain the account is as we still create a universal wallet